tests/Unit/Helpers/Evolution/DiscontinuousGalerkin/BoundaryCorrections.hpp

```diff
@@ -0,0 +1,317 @@
+      const auto& normal_dot_flux =
+          get<normal_dot_flux_tag>(interior_package_data);
+      auto& expected_boundary_correction =
+          get<dt_tag>(expected_boundary_corrections);
+      for (size_t tensor_index = 0;
+           tensor_index < expected_boundary_correction.size(); ++tensor_index) {
+        expected_boundary_correction[tensor_index] -=
+            normal_dot_flux[tensor_index];
+      }
+    });
+    {
+      INFO("Check weak and strong boundary terms match.");
+      CHECK(boundary_corrections == expected_boundary_corrections);
+    }
+
+    if (zero_on_smooth_solution == ZeroOnSmoothSolution::Yes) {
+      INFO(
+          "Testing that if the solution is the same on both sides the "
+          "StrongInertial correction is identically zero.");
+      Variables<dg_package_field_tags> interior_package_data_opposite_signs{
+          used_for_size.size()};
+      call_dg_package_data(make_not_null(&interior_package_data_opposite_signs),
+                           correction, interior_fields_on_face, volume_data,
+                           exterior_unit_normal_covector, mesh_velocity);
+      Variables<dt_variables_tags> zero_boundary_correction{
+          face_mesh.number_of_grid_points()};
+      call_dg_boundary_terms(make_not_null(&zero_boundary_correction),
+                             correction, interior_package_data,
+                             interior_package_data_opposite_signs,
+                             ::dg::Formulation::StrongInertial);
+      Variables<dt_variables_tags> expected_zero_boundary_correction{
+          face_mesh.number_of_grid_points(), 0.0};
+      tmpl::for_each<dt_variables_tags>([&expected_zero_boundary_correction,
+                                         &zero_boundary_correction](
+                                            auto dt_variables_tag_v) noexcept {
+        using dt_variables_tag = tmpl::type_from<decltype(dt_variables_tag_v)>;
+        const std::string tag_name = db::tag_name<dt_variables_tag>();
+        CAPTURE(tag_name);
+        Approx custom_approx = Approx::custom().epsilon(1.e-12).scale(1.0);
+        CHECK_ITERABLE_CUSTOM_APPROX(
+            get<dt_variables_tag>(zero_boundary_correction),
+            get<dt_variables_tag>(expected_zero_boundary_correction),
+            custom_approx);
+      });
+    }
+  } else if (dg_formulation == ::dg::Formulation::WeakInertial) {
+    INFO(
+        "Checking that swapping the two sides results in an overall minus "
+        "sign.");
+    Variables<dt_variables_tags> reverse_side_boundary_corrections{
+        face_mesh.number_of_grid_points()};
+    call_dg_boundary_terms(make_not_null(&reverse_side_boundary_corrections),
+                           correction, exterior_package_data,
+                           interior_package_data, dg_formulation);
+    // Reversing the sign and having them be equal is a statement of "what goes
+    // out one element enters the other".
```

Comment:
[clarity nitpick] "Reversing the sign and having them be equal" doesn't mean anything to me. I suggest this phrasing: "Check that the flux leaving one element equals the flux entering its neighbor, i.e., F*(interior, exterior) == -F*(exterior, interior)"